core/src/main/java/org/apache/carbondata/core/datastore/page/UnsafeVarLengthColumnPage.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.datastore.page;
+
+import java.math.BigDecimal;
+
+import org.apache.carbondata.core.memory.CarbonUnsafe;
+import org.apache.carbondata.core.memory.MemoryBlock;
+import org.apache.carbondata.core.memory.MemoryException;
+import org.apache.carbondata.core.memory.UnsafeMemoryManager;
+import org.apache.carbondata.core.metadata.datatype.DataType;
+import org.apache.carbondata.core.util.DataTypeUtil;
+
+// This extension uses unsafe memory to store page data, for variable length data type (string,
+// decimal)
+public class UnsafeVarLengthColumnPage extends VarLengthColumnPageBase {
+
+  // memory allocated by Unsafe
+  private MemoryBlock memoryBlock;
+
+  // base address of memoryBlock
+  private Object baseAddress;
+
+  // base offset of memoryBlock
+  private long baseOffset;
+
+  // size of the allocated memory, in bytes
+  private int capacity;
+
+  // default size for each row, grows as needed
+  private static final int DEFAULT_ROW_SIZE = 8;
+
+  UnsafeVarLengthColumnPage(DataType dataType, int pageSize) throws MemoryException {
+    super(dataType, pageSize);
+    capacity = pageSize * DEFAULT_ROW_SIZE;
+    memoryBlock = UnsafeMemoryManager.allocateMemoryWithRetry(capacity);
+    baseAddress = memoryBlock.getBaseObject();
+    baseOffset = memoryBlock.getBaseOffset();
+    rowOffset = new int[pageSize];
+    totolLength = 0;
+  }
+
+  @Override
+  public void freeMemory() {
+    if (memoryBlock != null) {
+      UnsafeMemoryManager.INSTANCE.freeMemory(memoryBlock);
+    }
+  }
+
+  private void ensureMemory(int requestSize) throws MemoryException {
+    if (totolLength + requestSize > capacity) {
+      memoryBlock = UnsafeMemoryManager.reallocateMemoryWithRetry(memoryBlock, requestSize);
+    }
```

Comment:
isn't it it should be like `UnsafeMemoryManager.reallocateMemoryWithRetry(memoryBlock, totolLength+requestSize)`